org.eclipse.debug.ui/ui/org/eclipse/debug/ui/InspectPopupDialog.java

```diff
@@ -166,11 +167,23 @@ public void updateComplete(IViewerUpdate update) {
 				}
 			}
 		});
+		if (doubleClickListener != null) {
+			fViewer.addDoubleClickListener(doubleClickListener);
+		}
 		fViewer.setInput(treeRoot);
 
 		return fTree;
 	}
 
+	/**
+	 * Adds the double click listener to the internal {@link TreeModelViewer}.
+	 *
+	 * @since 3.17
+	 */
+	public void addDoubleClickListener(IDoubleClickListener listener) {
```

Comment:
Why does it have to be public? Couldn't it be package visible, so it's not API?
If it's `add...` then user may expect multiple ones are allowed. Using `set...` would be better if you wish to stick to only 1 value.

Reply:
I want to use it in JDT, not from inside this package, so it's need to be public
Using set as the prefix makes sense, thanks!
I've added a check to verify that the internal tree was not initialized before. (Trying to support both scenario would make everything much more complex, I think).

Reply:
OK. and just to make sure it can't be a common feature here, what are you willing to add on double-click in JDT? I actually believe it would be better to keep `add...` and support multiple listeners, as it's done in SWT, JFace and about every other similar case.

Reply:
Yes, InspectPopupDialog could have a method to return the internal TreeModelViewer - the only thing, which is a bit strange with that, is that you have to call dialog.open() first, and only after could you add a listener to the viewer. But definitely a more flexible solution. That way multiple listeners, and everything else would be supported.
 I would like to navigate to the object types, and fields in JDT in the AllReferencesActionDelegate/AllInstancesActionDelegate, so for this, the only thing is to handle double clicks.

Reply:
I think `dialog.create()` as `dialog.open()` will block?

Reply:
I was surprised as well, but `InspectPopupDialog` inherits from `PopupDialog` and not from `Dialog`, so `open()` is not a blocking call